src/Utilities.js

```diff
@@ -55,26 +54,16 @@ export function walkDeepDescendantElements(root, callback) {
 
     // Ignore descendants of link elements to prevent attempting to traverse
     // elements created by the HTML Imports polyfill. When upgrading a tree,
-    // `CustomElementInternals#upgradeTree` by walks the `import` property of
-    // import links, which will point to either the true imported document
+    // `CustomElementInternals#patchAndUpgradeTree` walks the `import` property
+    // of import links, which will point to either the true imported document
     // (native) or the descendant containing the elements created to emulate the
     // imported document (polyfill).
     if (element.localName === 'link') {
-      let context = element;
-      let next = context.nextSibling;
-      while (!next && context !== root) {
-        context = context.parentNode;
-        next = context.nextSibling;
-      }
-
-      if (!next || context === root) return;
-
-      walker.currentNode = next;
-      walker.previousNode();
+      walker.currentNode = element.lastChild || element;
```

Comment:
Whoops, looks like this didn't work.  It'll still plunge into children of the link's last child, which is not what we want.  Sorry!!

This seems simpler than the old way:

```js
while (element.lastChild) {
  element = element.lastChild;
}
walker.currentNode = element;
```